modules/importer/src/runner/quay/walker.rs

```diff
@@ -0,0 +1,282 @@
+use super::oci::Reference;
+use crate::model::QuayImporter;
+use crate::runner::common::Error;
+use crate::runner::context::RunContext;
+use crate::runner::progress::{Progress, ProgressInstance};
+use crate::runner::quay::oci;
+use crate::runner::report::{Message, Phase, ReportBuilder};
+use futures::{Stream, StreamExt, future, stream};
+use reqwest::header;
+use serde::Deserialize;
+use std::{collections::HashMap, sync::Arc};
+use time::OffsetDateTime;
+use tokio::sync::Mutex;
+use tracing::instrument;
+use trustify_entity::labels::Labels;
+use trustify_module_ingestor::service::{Cache, Format, IngestorService};
+
+const QUAY_API_TOKEN: &str = "QUAY_API_TOKEN";
+
+#[derive(Clone, Debug, Default, serde::Serialize, serde::Deserialize)]
+pub struct LastModified(Option<i64>);
+
+pub struct QuayWalker<C: RunContext> {
+    continuation: LastModified,
+    importer: QuayImporter,
+    ingestor: IngestorService,
+    report: Arc<Mutex<ReportBuilder>>,
+    client: reqwest::Client,
+    context: C,
+}
+
+impl<C: RunContext> QuayWalker<C> {
+    pub fn new(
+        importer: QuayImporter,
+        ingestor: IngestorService,
+        report: Arc<Mutex<ReportBuilder>>,
+        context: C,
+    ) -> Result<Self, Error> {
+        let client = match std::env::var(QUAY_API_TOKEN) {
```

Comment:
Importers to have a configuration section. In cases we of system configuration, we use `clap`. So I am wondering why this token is fetched directly from an environment variable?

Also, having only one token would restrict us to only one quay instance, with only one quay "user" (token). What would happen if a user has two quay instances? What if there are two users/tokens with two different sets of permissions?

Reply:
Good points. I'll add the token to the importer config.